src/app/journey/JourneyContainer.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import {Component} from 'react';
+import {Inject} from 'react.di';
+import {DynamicContentHttpService} from "../dynamic-content/DynamicContentHttpService";
+import {JourneyData} from "./JourneyData";
+import {Journey} from "./Journey";
+
+interface JourneyContainerState {
+  content: JourneyData;
+}
+
+export class JourneyContainer extends Component<{}, JourneyContainerState> {
+
+  @Inject dynamicContentService: DynamicContentHttpService<JourneyData>;
+
+  private journeyContentKey: string = 'journey';
+
+  constructor(props) {
+    super(props);
+  }
```

Comment:
you should initialize state here, like `this.state = {}`; then you can omit `this.state`-check in the `render` function

Reply:
This doesn't work since {} is not of type JourneyContainerState.